### PR TITLE
Use Spock 2 consistently in build and documentation

### LINKED ISSUE #16359: Consistently use Spock2+JUnit platform in documentation and tests
After upgrading to Groovy 3, we have a mixture of Spock 2 + Junit 4 or Junit platform in our tests and documentation.

We should have a consistent way of using Spock 2

### PR DESCRIPTION
**Pull request by donat**
This PR addresses the following items:
- The TestKit snippets are updated to use JUnit 5
- Adjust URLs and convert them to use https
- Fix unresolved reference in `build_lifecycle.adoc`
- Remove the redundant test from `test_kit.adoc`: We listed the same Junit test snippet in the guide twice. The only difference was the implementation language, which is irrelevant in this chapter: it should only exhibit how to integrate with a particular test framework. 

I haven't looked at the samples yet; I'll create a separate PR for that if there's something to change.

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test rFM

**Comment by donat** [comment]
@bot-gradle test rFM

File: subprojects/docs/src/docs/userguide/reference/ci-systems/travis-ci.adoc [hunk 1]
```diff
@@ -18,12 +18,12 @@ In this guide, we'll discuss how to configure link:https://travis-ci.org/[Travis
 == Setup a typical project
 
 As example, this guide is going to focus on a Java-based project.
-More specifically, a Gradle plugin written in Java and tested with Spock.
+More specifically, a Gradle plugin written in Java and tested with https://www.spekframework.org/[Spek].
```
Comment (bmuskalla) [inline subprojects/docs/src/docs/userguide/reference/ci-systems/travis-ci.adoc:21]: Good catch :)

**Comment by donat** [comment]
@bot-gradle test rFR

File: subprojects/docs/src/snippets/testKit/gradleVersion/groovy/src/test/groovy/org/gradle/sample/BuildLogicFunctionalTest.groovy [hunk 2]
```diff
@@ -3,12 +3,12 @@ package org.gradle.sample
 // tag::functional-test-spock-gradle-version[]
 import org.gradle.testkit.runner.GradleRunner
 import static org.gradle.testkit.runner.TaskOutcome.*
-import org.junit.jupiter.api.io.TempDir;
+import spock.lang.TempDir
```
Comment (bmuskalla) [inline subprojects/docs/src/snippets/testKit/gradleVersion/groovy/src/test/groovy/org/gradle/sample/BuildLogicFunctionalTest.groovy:6]: I did not know!

File: subprojects/docs/src/snippets/testKit/junitQuickstart/groovy/build.gradle [hunk 3]
```diff
@@ -10,7 +10,12 @@ dependencies {
 
 // tag::declare-junit-dependency[]
 dependencies {
-    testImplementation 'junit:junit:4.13'
+    testImplementation("org.junit.jupiter:junit-jupiter-api:5.6.0")
```
Comment (bmuskalla) [inline subprojects/docs/src/snippets/testKit/junitQuickstart/groovy/build.gradle:13]: consider using the latest `5.7.1`

**Review by bmuskalla — APPROVED** [review]
LGTM

File: subprojects/docs/src/snippets/testKit/junitQuickstart/groovy/build.gradle (hunk 3 above)
Comment (donat) [reply subprojects/docs/src/snippets/testKit/junitQuickstart/groovy/build.gradle:13]: Thanks for the review. I'll look into this separately.

Comment (donat) [reply subprojects/docs/src/snippets/testKit/junitQuickstart/groovy/build.gradle:13]: Here's the follow-up PR: https://github.com/gradle/gradle/pull/16427

**Pull request merged**